core/services/ardupilot_manager/settings.py

```diff
@@ -0,0 +1,92 @@
+import json
+import pprint
+from copy import deepcopy
+from pathlib import Path
+from typing import Any, Dict
+from warnings import warn
+
+import appdirs
+
+
+class Settings:
+    app_name = "ardupilot-manager"
+    settings_path = Path(appdirs.user_config_dir(app_name))
+    settings_file = Path.joinpath(settings_path, "settings.json")
+    firmware_path = Path.joinpath(settings_path, "firmware")
+
+    def __init__(self) -> None:
+        self.root: Dict[str, Any] = {"version": 0, "content": {}}
+
+    def create_settings(self) -> None:
+        """Create settings files and folders."""
+        try:
+            Path.mkdir(self.settings_path, exist_ok=True)
+            Path.mkdir(self.firmware_path, exist_ok=True)
+            if not Path.is_file(self.settings_file):
+                with open(self.settings_file, "w+") as file:
+                    print(f"Creating settings file: {self.settings_file}")
+                    json.dump(self.root, file, sort_keys=True, indent=4)
+
+        except OSError as e:
+            warn(f"Could not create settings files: {e}\n No settings will be loaded or saved during this session.")
+
+    @property
+    def content(self) -> Any:
+        return self.root["content"]
+
+    @property
+    def version(self) -> Any:
+        return self.root["version"]
```

Comment:
version is an int, since this is know, the function should also return an int, otherwise every time that someone wants to use the function a cast will be necessary, we should unwrap unsafe stuff in the minimal scope possible.

Reply:
This and content where respectively str (i assume we won't have just int versions?) and Dict[str, Any], but I had complaints from MyPy from what seems to be a misunderstand from the library.
I will try to find a workaround.